app/Http/Controllers/Admin/InvitationCrudController.php

```diff
@@ -0,0 +1,130 @@
+    use \Backpack\CRUD\app\Http\Controllers\Operations\ShowOperation;
+
+    /**
+     * Configure the CrudPanel object. Apply settings to all operations.
+     * 
+     * @return void
+     */
+    public function setup()
+    {
+        CRUD::setModel(\App\Models\Invitation::class);
+        CRUD::setRoute(config('backpack.base.route_prefix') . '/invitation');
+        CRUD::setEntityNameStrings('invitation', 'invitations');
+    }
+
+    /**
+     * Define what happens when the List operation is loaded.
+     * 
+     * @see  https://backpackforlaravel.com/docs/crud-operation-list-entries
+     * @return void
+     */
+    protected function setupListOperation()
+    {
+        CRUD::setFromDb(); // columns
+
+        /**
+         * Columns can be defined using the fluent syntax or array syntax:
+         * - CRUD::column('price')->type('number');
+         * - CRUD::addColumn(['name' => 'price', 'type' => 'number']); 
+         */
+    }
+
+    /**
+     * Define what happens when the Create operation is loaded.
+     * 
+     * @see https://backpackforlaravel.com/docs/crud-operation-create
+     * @return void
+     */
+    protected function setupCreateOperation()
+    {
+        CRUD::setValidation(InvitationRequest::class);
+        // CRUD::setFromDb(); // fields
+        CRUD::field('name');
+        CRUD::field('email');
+        CRUD::addField(
+        [   // select2_from_array
+            'name'        => 'teams',
+            'label'       => "Teams",
+            'type'        => 'select2_from_array',
+            'options'     => $this->getTeams(),
+            'allows_null' => false,
+            'allows_multiple' => true, // OPTIONAL; needs you to cast this to array in your model;
+        ]
+        );
+
+
+
+
+        /**
+         * Fields can be defined using the fluent syntax or array syntax:
+         * - CRUD::field('price')->type('number');
+         * - CRUD::addField(['name' => 'price', 'type' => 'number'])); 
+         */
+    }
+
+    /**
+     * Define what happens when the Update operation is loaded.
+     * 
+     * @see https://backpackforlaravel.com/docs/crud-operation-update
+     * @return void
+     */
+    protected function setupUpdateOperation()
+    {
+        $this->setupCreateOperation();
+    }
+
+    public function store(InvitationRequest $request)
+    {
+        // do something before validation, before save, before everything
+        $response = $this->traitStore();
+        $invitation = $this->crud->getCurrentEntry();
+        $text_password = Str::random(8);
+        $password = Hash::make($text_password);
+      
+        $user = User::create([
+            'name' => $invitation['name'],
+            'email' => $invitation['email'],
+            'password'=> $password,
+
+        ]);
+        
+        
+        $user->save();
+        $user->teams()->sync($request['teams']); 
+        
+        Mail::to(backpack_user())->send(new InviteUser($user, $text_password));
```

Comment:
The email here is currently being sent to the user who creates the invite. Is this correct? Seems counter-intuitive to me, as I would assume the new user should receive their new login details directly. But this could be desired functionality - maybe the admin user should get the new user's login details and then pass them on manually? 

Let me know if this is intended here :)

Reply:
No, it is wrong. Sorry my bad